aptos-move/framework/aptos-framework/sources/configs/staking_config.move

```diff
@@ -109,8 +186,10 @@ module aptos_framework::staking_config {
         config.recurring_lockup_duration_secs
     }
 
+    /// DEPRECATING
     /// Return the reward rate.
     public fun get_reward_rate(config: &StakingConfig): (u64, u64) {
+        assert!(!features::reward_rate_decrease_enabled(), EDEPRECATED_FUNCTION);
```

Comment:
Aborting here will break all staking protocols that depend on this. How about just calculating the new rate and return it here?

Reply:
No let's make sure getter function doesn't have side effects